src/EditorFeatures/CSharpTest/MoveToNamespace/MoveToNamespaceTests.cs

```diff
@@ -1112,5 +1110,59 @@ class MyClass
 
             Assert.Empty(actions);
         }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.MoveToNamespace)]
+        [WorkItem(980758, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/980758")]
+        public Task MoveToNamespace_MoveOnlyTypeInGlobalNamespace()
+        => TestMoveToNamespaceAsync(
+@"class MyClass[||]
+{
+}",
+expectedMarkup: @"namespace {|Warning:A|}
+{
+    class MyClass
+    {
+    }
+}",
+targetNamespace: "A",
+expectedSymbolChanges: new Dictionary<string, string>()
+{
+    {"MyClass", "A.MyClass" }
+});
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.MoveToNamespace)]
+        [WorkItem(980758, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/980758")]
+        public async Task MoveToNamespace_MoveOnlyTypeToGlobalNamespace()
+        {
+            // We will not get "" as target namespace in VS, but the refactoring should be able
+            // to handle it w/o crashing.
+            await TestMoveToNamespaceAsync(
+
+  @"namespace A
+    class MyClass[||]
+    {
+    }
+}",
+  expectedMarkup: @"namespace A
+{
+    class MyClass
+    {
+    }
+}",
+      targetNamespace: "");
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.MoveToNamespace)]
+        [WorkItem(980758, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/980758")]
+        public Task MoveToNamespace_MoveOneTypeInGlobalNamespace()
+            => TestMoveToNamespaceAsync(
+@"class MyClass1[||]
```

Comment:
Why isn't this scenario supported?

Reply:
I think the change namespace service would take everything declared in global namespace in the file to the new namespace. It doesn't work on a single member basis.

Reply:
That said, we should file an issue to add the support for this

Reply:
Agreed. Definitely doesn't have to be fixed here, but it's odd it doesn't work. Please file an issue

Reply:
Filed #38767